doc/source/testingfeatures.rst

```diff
@@ -0,0 +1,80 @@
+====================
+Testing new features
+====================
+
+New features are constantly being developed for SunPy by contributors from
+around the world. These new features are normally submitted as a
+:ref:`Pull Request <dev-reference-label>`. Due to the review process imposed
+on pull requests to maintain the stability and consistency of the code base,
+it may take some time for a new feature to get accepted into the `master`
+branch, and a bit more to get into the next stable release version.
+
+In order to make it easier for people to test out new features (and hopefully provide valuable
+feedback to the authors) we've put together this tutorial.
+
+This assumes you have installed sunpy using anaconda, though a similar result
+can be obtained using `virtualenv`.
+
+We start by creating a new environment, so your current one still works as
+expected::
+
+ conda create -n sunpy-featureX python=3.5 sunpy
```

Comment:
I think Python 3 is more robust.

> If the user is currently running Python 2.7 and we are still partially supporting it, we should not "force" the user to a different version of Python, possibly triggering a whole raft of downloads for installation.

We are not "forcing" it, it's not that complex to see that you can swap a 3 for a 2. Also it makes people see that it's the default.

> If we've completely ditched all remnants of supporting Python 2.7, then there's no reason to have an extraneous argument to explicitly specify a Python version (and can only cause problems if there's a typo).

If we have dropped 2.7 then if someone has a python 2 conda install (but is using 3.x for sunpy in an env) then not setting python=3 will cause them to install python 2 and fail. Setting `python=3` is the only thing that will always work, now and in the future.